### PR TITLE
mesa: update to 20.3.1.

### PR DESCRIPTION
**Pull request by HadetTheUndying**
removed drm and surfaceless from platforms since they're determined automatically now.

### CODE REVIEW

File: srcpkgs/mesa/template [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 pkgname=mesa
-version=20.2.3
+version=20.3.1
 revision=1
 wrksrc="mesa-${version}"
 build_style=meson
 configure_args="-Dglvnd=true -Dshared-glapi=true -Dgbm=true -Degl=true
  -Dosmesa=gallium -Dgles1=true -Dgles2=true -Dglx=dri -Ddri3=true
- -Dlmsensors=true -Dplatforms=x11,drm,$(vopt_if wayland wayland,)surfaceless
+ -Dlmsensors=true -Dplatforms=x11,$(vopt_if wayland wayland)
```
Comment (ericonr) [inline srcpkgs/mesa/template:9]: Will this still build if one removes the wayland option?

**Review by ericonr — COMMENTED** [review]
Tested locally, the default build is working fine.

File: srcpkgs/mesa/template (hunk 1 above)
Comment (ericonr) [reply srcpkgs/mesa/template:9]: Yeah, it fails

> meson.build:21:0: ERROR: Options "" are not in allowed choices: "auto, x11, wayland, haiku, android, windows"

I'm okay with removing the build option :P

Comment (ericonr) [reply srcpkgs/mesa/template:9]: Otherwise, `x11$(vopt_if wayland ,wayland)`

**Comment by HadetTheUndying** [comment]
Done.

File: srcpkgs/mesa/template (hunk 1 above)
Comment (HadetTheUndying) [reply srcpkgs/mesa/template]: Woops commented in the wrong spot. Yeah i thought about that too. Went ahead and did it. Should be good now

File: srcpkgs/mesa/template [hunk 2]
```diff
@@ -23,7 +23,7 @@ license="MIT, LGPL-2.1-or-later"
 homepage="https://www.mesa3d.org/"
 changelog="https://docs.mesa3d.org/relnotes/${version}.html"
 distfiles="https://mesa.freedesktop.org/archive/mesa-${version}.tar.xz"
-checksum=ae1b240e11531df528d14dc214d2dc4d2b4f2e835c6230ba0b492b171eceb82b
+checksum=af751b49bb2ab0264d58c31e73d869e80333de02b2d1becc93f1b28c67aa780f
 
 build_options="wayland"
 build_options_default="wayland"
```
Comment (ericonr) [inline srcpkgs/mesa/template:29]: You need to remove the build option from here, then. Also from the dependencies.

Might be simpler to still use `vopt_if` :P

**Comment by q66** [comment]
we should fix the driver search path according to https://github.com/void-linux/void-packages/pull/27269 while at it

basically remove

```
case "$XBPS_TARGET_MACHINE" in
        i686) configure_args+=" -Ddri-drivers-path=/usr/lib32/dri";;
esac
```

and add `-Ddri-drivers-path=/usr/lib${XBPS_TARGET_WORDSIZE}` to `configure_args`

**Comment by travankor** [comment]
the transitional dummy packages can be removed, too (it's been  a year)

**Comment by HadetTheUndying** [comment]
Hey guys I'm going through a family crisis right now. Is there any way i can hand this off, or should I close my PR and let someone else submit a new one? I'm on mobile for the time being so i cannot make any of the suggested changes and don't want to hold things up.

Thanks

**Comment by ericonr** [comment]
@HadetTheUndying good luck out there.

Someone else can work on another PR that "closes" this one. Thank you for the work.

**Comment by HadetTheUndying** [comment]
Okay I will leave the PR open until another PR addresses it. If i have free time during the week and no one has gotten to it, I will wrap it up. Thank you.

**Comment by ericonr** [comment]
I opened #27291 that applies the comments from here.

**Comment by HadetTheUndying** [comment]
I'll update my 21.0.0 sometime this week with the updates. Closing now. Thanks.

**Pull request closed without merging**